timescaledb/tutorials/building-a-custom-timescaledb-dashboard.md

```diff
@@ -0,0 +1,326 @@
+* Mutation: convention for operations that are writing data. We’ll map the compression and decompression actions as mutations.
+
+### Queries and subscriptions
+
+Hasura has great flexibility to plug any resource and offer it as a query or a subscription. For this example, we’ll be mapping the *chunks_with_compression* view we created earlier as a GraphQL resource. It can be consumed as a query or subscription.
+
+![Tracking a SQL view on Hasura cloud](https://assets.timescale.com/docs/images/tutorials/visualizing-compression/hasura-cloud-track-view.png)
+
+We need to map those mutations to compress and decompress a chunk. To accomplish that, we need to take a look at how Hasura types work.
+
+### Mutations
+
+Hasura can easily map custom types that come from table structures. To create the necessary mutations, our functions should also return types that inherit from table structures.
+
+To easily create a new structure of the table from a query, we can call the query with limit 0:
+
+#### Compress chunk mutation
+
+```sql
+CREATE TABLE compressed_chunk AS
+SELECT compress_chunk((c.chunk_schema ||'.' ||c.chunk_name)::regclass)
+FROM   timescaledb_information.chunks c
+WHERE  NOT c.is_compressed limit 0;
+```
+
+Hasura needs some function to be tracked as a mutation. In this case, let's create the function to just rewrap the default compress_chunk from the TimescaleDB extension.
+
+Now, we can return the "compressed_chunk" in our function that will compress the chunk:
+```sql
+CREATE OR REPLACE FUNCTION compress_chunk_named(varchar) returns setof compressed_chunk AS $$
+  SELECT compress_chunk((c.chunk_schema ||'.' ||$1)::regclass)
+  FROM   timescaledb_information.chunks c
+  WHERE  NOT c.is_compressed
+  AND    c.chunk_name = $1 limit 1
+$$ LANGUAGE SQL VOLATILE;
+```
+
+Note that the function adds an extra where clause to not compress what is already compressed.
+
+![Tracking compress chunk mutation on Hasura cloud](https://assets.timescale.com/docs/images/tutorials/visualizing-compression/hasura-cloud-compress-chunk-mutation.png)
+
+#### Decompress chunk mutation
+
+We'll need a similar function for the decompression:
+
+```sql
+CREATE OR REPLACE FUNCTION decompress_chunk_named(varchar) returns setof compressed_chunk AS $$
+  SELECT decompress_chunk((c.chunk_schema ||'.' ||$1)::regclass)
+  FROM   timescaledb_information.chunks c
+  WHERE  c.is_compressed
+  AND    c.chunk_name = $1 limit 1
+$$ LANGUAGE SQL VOLATILE;
+```
+Now, the next step is to jump into the Hasura cloud and connect the database as a new data source.
+
+In the data panel, after setting up the PostgreSQL URI of your database, you can track each function as a query or mutation in a single click. Here is an example of compress_chunk_named function:
+
+In our case, the subscription goes to the chunks_with_compression, and here is what it looks like:
+
+You should also track decompress_chunk_named and compress_chunk_named as GQL mutations with a single argument.
```

Comment:
```suggestion
The next step is to go to the Hasura cloud and connect the database as a new data source. In the data panel, set up the PostgreSQL URI of your database, and then you can track each function as a query or mutation. This is an example of the `compress_chunk_named` function. In our case, the subscription goes to the  `chunks_with_compression` function. You can also track `decompress_chunk_named` and `compress_chunk_named` as GQL mutations with a single argument.
```
Missing images?

Reply:
oh yes, the images were a bit repetitive, then we have one less image here. Let me double-check in the final reading and I'll re-add if needed. As you rewrote these pieces now it looks awesome 🙇